src/accordion/accordion.tsx

```diff
@@ -19,21 +19,23 @@ const AccordionBase = ({
     "data-testid": testId,
     className,
     headingLevel = 2,
+    expandCollapseClickCallback,
 }: AccordionProps): JSX.Element => {
     const [expandAll, setExpandAll] = useState<boolean>(
         initialDisplay === "expand-all"
     );
 
-    const handleExpandCollapseClick = (event: React.MouseEvent) => {
+    const handleExpandCollapseClick = (event: React.MouseEvent, callback?: () => void) => {
         event.preventDefault();
         setExpandAll((prevExpandValue) => !prevExpandValue);
+        callback?.();
```

Comment:
actually since the callback is destructured, there's no need to pass it as an argument

this can directly be `onExpandCollapseAll?.()`